core/app/models/spree/unit_cancel.rb

```diff
@@ -29,6 +29,8 @@ def adjust!
   # This method is used by Adjustment#update to recalculate the cost.
   def compute_amount(line_item)
     raise "Adjustable does not match line item" unless line_item == inventory_unit.line_item
+    line_item.order.update!
+    line_item = Spree::LineItem.find(line_item.id)
```

Comment:
I hope this line is not necessary. This amounts to a line item reload, which would result in an out-of-sync object graph - something we've been working pretty hard to avoid.

Besides, I second @cbrunsdon in that this but sounds pretty important and deserves a spec. 

Good find!

Reply:
Thanks, I just came across it by chance !!
Also, I am not as good a Rails developer as you guys, so, I only understood like 10-20 % of what you guys are talking about, but I am glad that someone noticed it & is working on a fix for it. :)

Also, if you don't mind, @mamhoff could you please explain to me what is an out-of-sync object graph ? I heard this term for the first time.

Reply:
In almost every request, we load an order with line items, shipments, inventory units, and so on. When reloading anything other than the order, we have two copies of the same database entry in memory - that means the object graph, the network on in-memory-objects is not in sync.

Reply:
Hey @mamhoff thanks for the explanation. I had no clue of those things & I have been using Solidus for the past 8-9 months. :-P
I guess I still have way lot more to learn. Thanks :))